### DISCUSSION TITLE
I cannot clone codes via ssh.

### DISCUSSION THREAD

**Discussion by theodorehills** [community/community#89903] [New to GitHub] [answered]
I cannot clone codes by running `git clone git@github.com:my_username/my_repo.git` command. Each time I run that clone command, I got this error:
```plain
ssh: connect to host github.com port 22: Connection timed out
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Then I tested my connection to Github by running command `ssh -T git@github.com` and the result is
```plain
ssh: connect to host github.com port 22: Connection timed out
```

But these commands worked on another computer (nearly the same internet environment). This issue is happened on Windows.

**Comment by urdrops** [comment, marked as answer]
Verify that port 22 is not blocked by your network or ISP. Some networks restrict outgoing connections on port 22. If necessary, you can try using a different network or contact your network administrator to allow access to port 22.

**Reply by theodorehills** [reply]
Thank you for this advice, I already checked the network situation, the connection to github is banned by local administrations.

**Comment by urdrops** [comment]
If the connection to GitHub is banned by local administrations or if port 22 is blocked, you might need to use alternative methods to access GitHub repositories. As an alternative, you can use the HTTPS URL for your Git remote instead of the SSH URL. Update the remote URL using the following command: 
```git remote set-url origin https://github.com/username/repository.git```

**Reply by theodorehills** [reply]
HTTPS and SSH are both banned, I tried using proxy with tunnel which is able to connect github via ssh.
